store/tracekvstore.go

```diff
@@ -0,0 +1,128 @@
+package store
+
+import (
+	"encoding/json"
+	"io"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	writeOp operation = "W"
+	readOp  operation = "R"
+)
+
+type (
+	// TraceKVStore implements the KVStore interface with tracing enabled.
+	// Operations are traced on each core KVStore call and written to the
+	// underlying io.writer.
+	//
+	// TODO: Should we use a buffered writer and implement Commit on
+	// TraceKVStore?
+	TraceKVStore struct {
+		parent sdk.KVStore
+		writer io.Writer
+	}
+
+	// operation represents an IO operation
+	operation string
+
+	// traceOperation implements a traced KVStore operation
+	traceOperation struct {
+		Timestamp int       `json:"timestamp,omitempty"`
```

Comment:
I think we want to log block height, not timestamp.

Reply:
@cwgoes suggestions about how to go about doing this? Do we need to pass/use the context?